File: python/1047.py
initialHour, initialMinutes, endHour, endMinutes = input().split()
initialHour = int(initialHour)
initialMinutes = int(initialMinutes)
endHour = int(endHour)
endMinutes = int(endMinutes)

totalHours = 0
totalMinutes = 0

if endHour < initialHour :
    totalHours = (24 - initialHour) + endHour
elif endHour == initialHour:
    totalHours = 24
else:
    totalHours = endHour - initialHour
    
if endMinutes > initialMinutes:
    if endHour == initialHour:
        totalHours = 0
    totalMinutes = endMinutes - initialMinutes
elif initialMinutes > endMinutes:
    totalHours -= 1;
    totalMinutes = 60 - (initialMinutes - endMinutes)
    
print(f'O JOGO DUROU {totalHours} HORA(S) E {totalMinutes} MINUTO(S)')

File: lua/1047.lua
local entry = io.read()
local t = {}
for v in string.gmatch(entry, '[^%s]+') do
    table.insert(t, tonumber(v))
end

local initialHour = t[1]
local initialMinute = t[2]
local finalHour = t[3]
local finalMinute = t[4]

local totalInitialMinutes = initialHour * 60 + initialMinute
local totalFinalMinutes = finalHour * 60 + finalMinute

local minutesDiff = totalFinalMinutes - totalInitialMinutes
if minutesDiff <= 0 then
    minutesDiff = 1440 + minutesDiff
end

print('O JOGO DUROU '..string.format('%i', minutesDiff // 60)..' HORA(S) E '..string.format('%i', minutesDiff % 60)..' MINUTO(S)')